src/vm/callingconvention.h

```diff
@@ -1016,7 +1016,7 @@ int ArgIteratorTemplate<ARGITERATOR_BASE>::GetNextOffset()
     case ELEMENT_TYPE_VALUETYPE:
     {
 #ifdef UNIX_AMD64_ABI
-        MethodTable *pMT = m_argTypeHandle.AsMethodTable();
+        MethodTable *pMT = m_argTypeHandle.GetMethodTable();
```

Comment:
@luqunl can you please educate me on why was this change needed?

Reply:
For some cases, m_argTypeHandle will point to typepamaterdesc instead of methodtable.

Reply:
I saw this comment in typehandle.h
```
    // Unlike AsMethodTable, GetMethodTable will get the method table
    // of the type, regardless of whether it is an array etc. Note, however
    // this method table may be shared, and some types (like TypeByRef), have
    // no method table (and this function returns NULL for them)
    inline PTR_MethodTable GetMethodTable() const;
```